src/libraries/System.Private.CoreLib/src/System/Version.cs

```diff
@@ -304,10 +319,18 @@ public static bool TryParse([NotNullWhen(true)] string? input, [NotNullWhen(true
         public static bool TryParse(ReadOnlySpan<char> input, [NotNullWhen(true)] out Version? result) =>
             (result = ParseVersion(input, throwOnFailure: false)) != null;
 
-        private static Version? ParseVersion(ReadOnlySpan<char> input, bool throwOnFailure)
+        public static bool TryParse(ReadOnlySpan<byte> utf8Text, [NotNullWhen(true)] out Version? result) =>
+            (result = ParseVersion(utf8Text, throwOnFailure: false)) != null;
```

Comment:
nit: This is a one liner, but it is visually difficult to process. We tend to prefer more readable code, even if more verbose, where possible.

I think this would be better simply as:
```csharp
{
    result = ParseVersion(utf8Text, throwOnFailure: false);
    return result is not null;
}
```

-------------

More generally, the shape of the internal `ParseVersion(utf8Text, throwOnFailure: false)` API is fairly atypical compared to the rest of our parsing/formatting code.

It would be nice if this were instead the more standard `Parse` calls `TryParse` and throws if the `bool` result is `false` pattern instead, but I think that's better for a separate PR

Reply:
> nit: This is a one liner, but it is visually difficult to process. We tend to prefer more readable code, even if more verbose, where possible.
I followed the pattern in existing method. Changed all of them now.